### DISCUSSION TITLE
One components inside one other

### DISCUSSION THREAD

**Discussion by nervusdm** [vuejs/petite-vue#213] [Q&A]
Hello,
I am familiar with vue.js. (not an expert, but..).
How can I use a component "myCoolButton" from a file1.html inside file2.html (for example).

I tried several things, without succes.

**Comment by VasylHulpak** [comment]
Hello @nervusdm 

You need to use slots for it. I guess it is the best solution(https://vuejs.org/guide/components/slots.html#slot-content-and-outlet)

**Comment by nervusdm** [comment]
Hello @VasylHulpak Could you, if it does not bother you, give me a quick example in petite vue ?

**Reply by VasylHulpak** [reply]
Hello, sure. In few minutes i will give you the example

**Reply by VasylHulpak** [reply]
@nervusdm 
There is not implementation yet. 
I saw only a discussion. I am working on another stuff so will not be able to fork the repo to implement. 

Sorry for that

**Comment by sqllyw** [comment]
> Hello, I am familiar with vue.js. (not an expert, but..). How can I use a component "myCoolButton" from a file1.html inside file2.html (for example).
> 
> I tried several things, without succes.

in petite-vue, component is just a function, so you put myCoolButton in a javascript file, then include it in file2.html's script part

**Reply by nervusdm** [reply]
I may be wrong, but not really,

Look about slot.

**Reply by sqllyw** [reply]
there is no slot in petite-vue, only a bare-bone component:
https://github.com/vuejs/petite-vue#components

**Reply by jikkuatwork** [reply]
Thats sad. Wish there was a way to nest components
